syntax = "proto3";

package street_manata;

message GpsPosition {
  float longitude = 1;
  float latitude = 2;
  float elevation = 3;
  optional float position_accuracy = 4;
  optional float elevation_accuracy = 5;
  optional float speed = 6;
  optional float speed_accuracy = 7;
  optional float heading = 8;
  optional float heading_accuracy = 9;
}

message Orientation {
  float roll = 1;
  float pitch = 2;
  float yaw = 3;
}

message Acceleration {
  float x = 1;
  float y = 2;
  float z = 3;
}

message AngularVelocity {
  float x = 1;
  float y = 2;
  float z = 3;
}

message MagneticField {
  float x = 1;
  float y = 2;
  float z = 3;
}

message OrientationCapture {
  int64 epoch_micro_seconds = 1;
  Orientation orientation = 2;
}

message GpsCapture {
  int64 epoch_micro_seconds = 1;
  GpsPosition position = 2;
}

message AccelerationCapture {
  int64 epoch_micro_seconds = 1;
  Acceleration acceleration = 2;
}

message AngularVelocityCapture {
  int64 epoch_micro_seconds = 1;
  AngularVelocity angular_velocity = 2;
}

message MagneticFieldCapture {
  int64 epoch_micro_seconds = 1;
  MagneticField magnetic_field = 2;
}

message CameraSpecification {
  optional int32 camera_index = 1;
  optional float focal_length = 2;
  optional float exposure_time = 3;
}

message PhotoCapture {
  int64 epoch_micro_seconds = 1;
  string file = 2;
  optional CameraSpecification camera_specification = 3;
  optional GpsPosition gps_position = 4;
  optional Orientation orientation = 5;
}

message VideoCapture {
  int64 start_epoch_micro_seconds = 1;
  int64 end_epoch_micro_seconds = 2;
  string file = 3;
  optional int64 fps = 4;
  optional CameraSpecification camera_specification = 5;
}

message GeoCapture {
  string device = 1;
  repeated GpsCapture gps = 2;
  repeated OrientationCapture orientation = 3;
  repeated AccelerationCapture acceleration = 4;
  repeated AngularVelocityCapture angular_velocity = 5;
  repeated MagneticFieldCapture magnetic_field = 6;
  repeated PhotoCapture photos = 7;
  repeated VideoCapture videos = 8;
  optional string description = 9;
  repeated string tags = 10;
}
